server/src/github.com/alexeyknyshev/cpsrv/cpsrv.go

```diff
@@ -130,47 +111,20 @@ func getRequestJsonStr(r *http.Request, context string) (string, error) {
 	return string(jsonStr), nil
 }
 
-func prepareResponse(w http.ResponseWriter, r *http.Request) (bool, int64) {
-	requestId, err := getRequestUserId(r)
-	if err != nil {
-		log.Printf("%s prepareResponse %v\n", getRequestContexString(r), err)
-		w.WriteHeader(http.StatusBadRequest)
-		return false, 0
-	}
-	if requestId == 0 {
-		log.Printf("%s prepareResponse unexpected requestId: %d\n", getRequestContexString(r), requestId)
-		w.WriteHeader(http.StatusBadRequest)
-		return false, 0
-	}
-
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	w.Header().Set("Id", strconv.FormatInt(requestId, 10))
-	return true, requestId
-}
-
-func writeResponse(w http.ResponseWriter, r *http.Request, requestId int64, responseBody string) {
-	io.WriteString(w, responseBody)
-	go logResponse(w, r, requestId, responseBody)
-}
-
-func writeHeader(w http.ResponseWriter, r *http.Request, requestId int64, code int) {
-	w.WriteHeader(code)
-	go logResponse(w, r, requestId, "code " + strconv.FormatInt(int64(code), 10))
-}
-
```

Comment:
Everything above must be restored with `go logResponse(w, r, requestId, responseBody)` replaced by logger (looks like you should pass logger as additional paramenter to each function call, i.e:
`writeResponse(logger, w, r, requestId, jsonStr)`  - just example )